test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -331,6 +460,11 @@ public void testAliasCreatedForCtorDepth1_1() {
     test("/** @constructor */ var a = function(){}; a.b = 1; "
          + "var c = a; c.b = 2; a.b != c.b;",
          "var a = function(){}; var a$b = 1; var c = a; c.b = 2; a$b != c.b;");
+
+    // Sometimes we want to prevent static members of a constructor from
+    // being collapsed.
+    testSame("/** @constructor */ var a = function(){};"
+        + "/** @nocollapse */ a.b = 1; var c = a; c.b = 2; a.b == c.b;");
```

Comment:
I think the answer should be no we don't. From a user perspective, I'd like `@nocollapse` to be analogous to `@export` which only applies to the immediate property. If sub-properties need protected from collapsing, then they would also need `@nocollapse`.

I'd like to prevent the issue that occurs when an `@nocollapse` annotation on a top-level property blocks optimizations on entire chunks of code. I specifically wrote it to apply to only the associated property.